flow/connectors/postgres/postgres.go

```diff
@@ -771,11 +772,48 @@ func (c *PostgresConnector) GetTableSchema(
 	return res, nil
 }
 
+func (c *PostgresConnector) GetSelectedColumns(
+	ctx context.Context,
+	sourceTable *utils.SchemaTable,
+	excludedColumns []string,
+) ([]string, error) {
+	// Given the list of excluded columns,
+	// we need to get the list of columns that are not excluded
+	getColumnsSQL := "SELECT attname AS column_name, format_type(atttypid, atttypmod) AS data_type, attnotnull AS is_nullable FROM pg_attribute WHERE attrelid = '%s.%s'::regclass AND attnum > 0 AND NOT attisdropped"
```

Comment:
can't you just pass the excluded columns into the query and filter on `attname`? handling empty array of course